### DISCUSSION TITLE
Github actions eslint: command not found

### DISCUSSION THREAD

**Discussion by PulkitBanta** [community/community#26593] [Actions]
<p>I recently created this action and want to lint the typescripts files using the eslint, but I get the error saying that <code>eslint: command not found</code>.</p>
<p>I am not sure what is wrong here, I need help to fix this issue.</p>
<h3>This is my workflow file</h3>
<pre><code class="lang-auto">name: CI

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - name: Install modules
      run: yarn
    - name: Run ESLint
      run: eslint . --ext .ts,.tsx
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/f/8/f85695468b0d9a06687e6585f76d609dadbd389b.png" data-download-href="/uploads/short-url/zqTUNzZK9aPZbVN0QzZSCaKZfd1.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/43134750/181098603-cf6901bc-41fc-4a4f-b652-57985286ffad.png" alt="image" data-base62-sha1="zqTUNzZK9aPZbVN0QzZSCaKZfd1" width="690" height="209" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/f/8/f85695468b0d9a06687e6585f76d609dadbd389b_2_690x209.png, https://aws1.discourse-cdn.com/github/optimized/3X/f/8/f85695468b0d9a06687e6585f76d609dadbd389b_2_1035x313.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/f/8/f85695468b0d9a06687e6585f76d609dadbd389b_2_1380x418.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/f/8/f85695468b0d9a06687e6585f76d609dadbd389b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1884×572 22 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Ic3m4n34** [comment]
<p>I’m having the same issue. Could you figure that one out by now?</p>

**Comment by technoir9** [comment, marked as answer]
<p>I had the same issue, adding <code>yarn run</code> before <code>eslint</code> fixed it for me.</p>
<p>In case of your file, you’d have to change the last line to:</p>
<pre><code class="lang-auto">      run: yarn run eslint . --ext .ts,.tsx
</code></pre>
<p>If you’re using <code>npm</code> then you should replace <code>yarn run</code> with <code>npx</code>.</p>

**Reply by RobinClowers** [reply]
why in the world would this be the case?

**Reply by tiennguyen1293** [reply]
Why can't we use yarn lint as defined in the scripts?

**Reply by ScarpMetal** [reply]
@RobinClowers With `yarn run` it uses the local dependencies in node_modules (dependencies installed via the `yarn` command). Without `yarn run` it looks for global dependencies (dependencies installed to the whole machine) which don't exist.

@tiennguyen1293 Yeah if you already had a "lint" script in your package.json you could just use `yarn run lint`

**Comment by PulkitBanta** [comment]
<p>Thanks for the solution it worked :slight_smile:</p>

**Comment by dmartines** [comment]
<p>Alternatively you can run using <code>./node_modules/.bin/eslint</code></p>

**Comment by tiennguyen1293** [comment]
package.json

```
{
  ...
  "scripts": {
    "lint": "yarn run eslint src --ext js,jsx",
  }
}
```

ci.yml
```
...
      - run: yarn install
        env:
          NODE_AUTH_TOKEN: ${{ secrets.NPM_REGISTRY_TOKEN }}
      - run: yarn lint
```


And i'm facing this issue

![image](https://github.com/community/community/assets/26110369/14d5f7fa-0f94-4b22-9126-752a686f6058)

Q: How can I use a package script to run `eslint`?

Thanks

**Reply by ScarpMetal** [reply]
Make sure you have eslint installed in your devDependencies
